optax/contrib/_muon.py

```diff
@@ -0,0 +1,254 @@
+"""Muon.
+
+Implementation of the
+[Muon optimizer](https://github.com/KellerJordan/modded-nanogpt)
+by Keller Jordan
+"""
+
+
+from typing import Any, NamedTuple, Optional, Tuple, Union
```

Comment:
nit: could you remove `Optional, Tuple, Union` and use python 3.9+ annotations, e.g., Union[A, B] === A | B, Optional[A] === A | None